.dotnet/Azure.AI.OpenAI.Staging/Custom/AzureOpenAIClient.cs

```diff
@@ -0,0 +1,281 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.AI.OpenAI.Staging.Audio;
+using Azure.AI.OpenAI.Staging.Chat;
+using Azure.AI.OpenAI.Staging.Embeddings;
+using Azure.AI.OpenAI.Staging.FineTuning;
+using Azure.AI.OpenAI.Staging.Images;
+using Azure.Core;
+using OpenAI;
+using OpenAI.Assistants;
+using OpenAI.Audio;
+using OpenAI.Chat;
+using OpenAI.Embeddings;
+using OpenAI.Files;
+using OpenAI.FineTuning;
+using OpenAI.Images;
+using OpenAI.ModelManagement;
+using OpenAI.Moderations;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Azure.AI.OpenAI.Staging;
+
+/// <summary>
+/// The top-level client for the Azure OpenAI service.
+/// </summary>
+/// <remarks>
+/// For scenario-specific operations, create a corresponding client using the matching method on this type, e.g. <see cref="GetChatClient(string)"/>.
+/// </remarks>
+public partial class AzureOpenAIClient : OpenAIClient
+{
+    private const string azureOpenAIEndpointEnvironmentVariable = "AZURE_OPENAI_ENDPOINT";
+    private const string azureOpenAIApiKeyEnvironmentVariable = "AZURE_OPENAI_API_KEY";
+
+    private readonly AzureOpenAIClientOptions _options;
+
+    internal static PipelineMessageClassifier PipelineMessageClassifier200 => _pipelineMessageClassifier200.Value;
+    internal static Lazy<PipelineMessageClassifier> _pipelineMessageClassifier200
```

Comment:
IIRC using Lazy costs an allocation -- consider making the field private and nullable?  I think there should be examples floating around in generated DPG clients -- lmk if it would be helpful to dig up a link.

Reply:
Oh, it does? Had no idea. Yes, the pattern was already there for the taking -- I was just being stubborn. I'll replace it in both places.